### DISCUSSION TITLE
weird processes of webview2

### DISCUSSION THREAD

**Discussion by yzqdev** [wailsapp/wails#1509] [Q&A] [answered]
i use wails to build an app. In production, when i open the file generated by `wails build`, sometime it has only one process(itself) with very little memory, sometimes it has a lot webview2 processes with much memory. (the exe build by the default template did the same)

I don't know why. is there anyway we could avoid it?

only one process

![image](https://user-images.githubusercontent.com/20037151/176615446-7b781674-0421-4d5a-b3ab-ccc0fda491fe.png)

with many webview2 processes

![image](https://user-images.githubusercontent.com/20037151/176615686-d19e8ba7-bb2f-491e-93c2-a6c4b8a373f0.png)

default template
![image](https://user-images.githubusercontent.com/20037151/176619509-2053c17c-572b-4855-a440-9048eea751a9.png)

**Comment by stffabi** [comment, marked as answer]
Wails internally uses WebView2 of Microsoft to render the HTML frontend and that runs out of the main process. That's why you see those WebView2 processes and they are just normal. It's an essential component of Wails on Windows.

Sometimes the task manager fails to group them together correctly. IIRC this happens when the task manager is already running when starting the Wails application.
